src/charm.py

```diff
@@ -276,14 +277,10 @@ def _on_loki_pebble_ready(self, _):
             logger.debug("Cannot set workload version at this time: could not get Loki version.")
 
     def _on_node_exporter_pebble_ready(self, _):
-        current_layer = self._node_exporter_container.get_plan()
         new_layer = self._node_exporter_pebble_layer
-        restart = current_layer.services != new_layer.services
-
-        if restart:
-            self._node_exporter_container.add_layer("node-exporter", new_layer, combine=True)
-            self._node_exporter_container.restart("node-exporter")
-            logger.info("Node Exporter (re)started")
+        self._node_exporter_container.add_layer("node-exporter", new_layer, combine=True)
+        self._node_exporter_container.replan()
```

Comment:
I'm missing some context on this obviously, but why is it that we can't add startup=True to the layer definition instead?

Reply:
can we have clarity on the reason, before we spend more time solving unknown issues?